File: ResponseChat.php
<?php

class ResponseChat {

        protected $data = array();
        private $btns ;
        private $logger;

        public function __construct($loggerFileName = null)
        {
                global $logger; 
                $this->logger = $logger;//new Logger($loggerFileName);
                $this->data = array("bot_state"=> time() );
        }
        public function setBotState($state=""){
                $this->data['bot_state'] = $state;
        }

        public function sendText($msg=""){
                $this->logger->debug(print_r(get_defined_vars(), true));

                $this->data['data'][] = array("type" => "text", "body"=>array("data" => $msg) );

        }
        public function sendDocument($link="", $filename="",  $mimeType="", $caption="" ){
                $this->logger->debug(print_r(get_defined_vars(), true));

                $this->data['data'][] = array("type" => "document", "body" => array( "link"=>$link,  "filename" => $filename,"mime_type" => $mimeType,"caption" => $caption));
                //$this->send();
        }
        public function sendImage($link="",$mimeType="", $caption="" ){
                $this->data['data'][] = array("type" => "image", "body" => array( "link"=>$link, "mime_type" => $mimeType,"caption" => $caption ) );
        }
        public function sendAudio($link="",$mimeType="", $caption="" ){
                $this->data['data'][] = array("type" => "audio", "body" => array( "link"=>$link, "mime_type" => $mimeType,"caption" => $caption ) );
        }
        public function sendLocation($latitude="",$longitude="", $name="", $address= "" ){
                $this->data['data'][] = array("type" => "location", "body" => array( "latitude"=>$latitude, "longitude" => $longitude,"name" => $name, "address" => $address ) );
        }
        public function sendVideo($link="",$mimeType="", $caption="" ){
                $this->data['data'][] = array("type" => "video", "body" => array( "link"=>$link, "mime_type" => $mimeType,"caption" => $caption ) );
        }
        public function addContact($contatObject ){
                $this->data['data'][] = array("type" => "Contact", "Contact"=> $contatObject);
        }
		public function sendAgentTransfer($skillName="", $uui="" ){
			$this->logger->debug(print_r(get_defined_vars(), true));

			$this->data['data'][] = array("type" => "CCTransfer", "skillName"=>$skillName, "uui" => $uui);
	}

	public function initInteractive($name) {
			$list = new listObject($name);
			return $list;
	}

	public function addInteractiveObject( $listObject  ){
			$this->logger->debug(print_r(get_defined_vars(), true));
			$this->data['data'][] = array("type" => "interactive", "body" => $listObject );
	}

	public function addButtons( $listObject  ){
			$this->logger->debug(print_r(get_defined_vars(), true));
			$this->data['data'][] = array("type" => "interactive", "body" => $listObject );
	}

	public function getData(){
			return  json_encode($this->data);
	}


	public function getResponse(){
			return  $this->data;
	}
	public function putTestData($data){
			$this->data = $data;
	}
	// Parse the XML.and Deconstruct
	public function getXML() {
			return $this->doc->saveXML();
	}
	public function printData() {
			//          header("Content-Type: application/json");
			return json_encode($this->data);
	}
	public function send(){
			header("Content-Type: application/json");
			$this->logger->info("whatsapp post data:::".json_encode($this->data,JSON_PRETTY_PRINT));
			echo $this->getData();   //  json_encodei($this->data);
			//exit;

	}

}
class Contact {
	private $type; //interactive
	protected $contact;
	protected $text;


	public function __construct($link=''){
			$this->text = "Please share below datails";
			$this->link = $link;
			$this->contact = []; //default empty
	}

	public function addText($key){
			$this->text=$key;
	}

	public function addName($name){
			$Object = array("key"=>"name","value"=>$name);
			array_push($this->contact,$Object);
	}
	public function addPhone($phone){

			$Object = array("key"=>"phone","value"=>$phone);
			array_push($this->contact,$Object);


	}
	public function addEmail($email){
			$Object = array("key"=>"email","value"=>$email);
			array_push($this->contact,$Object);
	}


	public function addAddress($address){
			$Object = array("key"=>"address","value"=>$address);
			array_push($this->contact,$Object);
	}
	public function addCustomField($key,$value){
			$Object = array("key"=>$key,"value"=>$value);
			array_push($this->contact,$Object);
	}

	public function returnContact(){
			$payload = [
					"text" => $this->text,
					"fields"=>  $this->contact,
			];
			return $payload;
	}

}

class Menu {


	private $type; //interactive
	private $title;
	protected $buttons;

	public function __construct($type,$title,$link){
			$this->type = $type;
			$this->title = $title;
			$this->link = $link;
			$this->buttons = []; //default empty
	}

	public function addChoice($id,$title){
			$buttonObject = array("id"=>$id,"title"=>$title);
			array_push($this->buttons,$buttonObject);

	}
	public function returnMenuPayload(){

			$payload = [
					"type" => $this->type,
					"title"=>  $this->title,
					"link"=>$this->link,
					"choices"=> $this->buttons
			];
			return $payload;
	}

	public function returnButtonPayloadImage($image){

			$payload = [
					"type" => $this->type,
					$this->type => [
							"type"=> "button",
					"header"=>[
							"type"=>"image",
					"image"=>[
							"link"=>$image
					]
					],
					"body"=> [
							"text" => $this->title,
					],
					"action"=> [
							"buttons"=>$this->buttons
					]

					]
			];
			$this->buttons = []; //clear buttons
			return $payload;


	}

}
class buttonObject {
	private $header;
	private $body;
	private $footer;
	private $choices;

	public function __construct($title){
			$this->body = $title;
			$this->choices = array();
	}

	public function addButton($id, $title){
			$this->choices[] = array("id"=> $id, "title"=>$title);
	}

	public function buildObject(){
			$payload = [
					"type" => "button",
					"title"=>  $this->body,
					"choices"=> $this->choices
			];
			return $payload;
	}
}



#*************** List Object ***********************

class listObject {
	private $header;
	private $body;
	private $footer;
	private $sections;

	public function __construct($title){
			$this->body = $title;
			$this->sections = array();
	}

	public function Section($name) {
			$section = new listObjectSection($name);
			return $section;
	}

	public function Button($name) {
			$section = new buttonObject($name);
			return $section;
	}

	public function addSection($section){
			$this->sections[] = $section;
	}


	public function buildObject(){
			$payload = [
					"type" => "list",
					"title"=>  $this->body,
					"sections"=> $this->sections
			];
			return $payload;
	}
}
class listObjectSection {
	private $title;
	private $choices = [];

	public function __construct($title) {
			$this->title = $title;
	}

	public function addChoice($id, $title, $description="") {
			// $choice = new Choice($name, $value);
			$this->choices[] = array("id"=>$id,"title"=>$title,"description"=>$description);
			return $this;
	}

	public function buildObject() {
			return array("title" => $this->title, "choices" => $this->choices);
	}

}

File: ResponseChat.py
import json
from time import time

class ResponseChat:
    def __init__(self, loggerFileName=None):
        self.data = {"bot_state": str(time()), "data": []}
        self.logger = Logger(loggerFileName) if loggerFileName else None

    def set_bot_state(self, state=""):
        self.data['bot_state'] = state

    def send_text(self, msg=""):
        self.logger.debug(locals()) if self.logger else None
        self.data['data'].append({"type": "text", "body": {"data": msg}})

    def send_document(self, link="", filename="", mimeType="", caption=""):
        self.logger.debug(locals()) if self.logger else None
        self.data['data'].append({"type": "document", "body": {"link": link, "filename": filename, "mime_type": mimeType, "caption": caption}})

    def send_image(self, link="", mimeType="", caption=""):
        self.data['data'].append({"type": "image", "body": {"link": link, "mime_type": mimeType, "caption": caption}})

    def send_audio(self, link="", mimeType="", caption=""):
        self.data['data'].append({"type": "audio", "body": {"link": link, "mime_type": mimeType, "caption": caption}})

    def send_location(self, latitude="", longitude="", name="", address=""):
        self.data['data'].append({"type": "location", "body": {"latitude": latitude, "longitude": longitude, "name": name, "address": address}})

    def send_video(self, link="", mimeType="", caption=""):
        self.data['data'].append({"type": "video", "body": {"link": link, "mime_type": mimeType, "caption": caption}})

    def add_contact(self, contact_object):
        self.data['data'].append({"type": "Contact", "Contact": contact_object})

    def send_agent_transfer(self, skillName="", uui=""):
        self.logger.debug(locals()) if self.logger else None
        self.data['data'].append({"type": "CCTransfer", "skillName": skillName, "uui": uui})

    def init_interactive(self, name):
        return ListObject(name)

    def add_interactive_object(self, list_object):
        self.logger.debug(locals()) if self.logger else None
        self.data['data'].append({"type": "interactive", "body": list_object.build_object()})

    def add_buttons(self, list_object):
        self.logger.debug(locals()) if self.logger else None
        self.data['data'].append({"type": "interactive", "body": list_object.build_object()})

    def get_data(self):
        return json.dumps(self.data)

    def get_response(self):
        return self.data

    def put_test_data(self, data):
        self.data = data

    def get_xml(self):
        # Code for generating XML representation of data
        pass

    def print_data(self):
        return json.dumps(self.data)

    def send(self):
        self.logger.info("whatsapp post data:::" + json.dumps(self.data, indent=4)) if self.logger else None
        print(self.get_data())


class Menu:
    def __init__(self, type, title, link):
        self.type = type
        self.title = title
        self.link = link
        self.buttons = []

    def add_choice(self, id, title):
        self.buttons.append({"id": id, "title": title})

    def return_menu_payload(self):
        payload = {
            "type": self.type,
            "title": self.title,
            "link": self.link,
            "choices": self.buttons
        }
        return payload

    def return_button_payload_image(self, image):
        payload = {
            "type": self.type,
            self.type: {
                "type": "button",
                "header": {
                    "type": "image",
                    "image": {
                        "link": image
                    }
                },
                "body": {
                    "text": self.title
                },
                "action": {
                    "buttons": self.buttons
                }
            }
        }
        self.buttons = []
        return payload


class ButtonObject:
    def __init__(self, title):
        self.body = title
        self.buttons = []

    def add_button(self, id, title):
        self.buttons.append({"id": id, "title": title})

    def build_object(self):
        payload = {
            "type": "button",
            "title": self.body,
            "choices": self.buttons
        }
        return payload


class ListObject:
    def __init__(self, title):
        self.body = title
        self.sections = []

    def section(self, title):
        section = ListObjectSection(title)
        self.sections.append(section)
        return section

    def build_object(self):
        payload = {
            "type": "list",
            "title": self.body,
            "sections": [section.build_object() for section in self.sections]
        }
        return payload


class ListObjectSection:
    def __init__(self, title):
        self.title = title
        self.choices = []

    def add_choice(self, id, title, description=""):
        self.choices.append({"id": id, "title": title, "description": description})

    def build_object(self):
        payload = {
            "title": self.title,
            "choices": self.choices
        }
        return payload